src/braintree/initBraintreePayPalButtons.ts

```diff
@@ -0,0 +1,310 @@
+import { getCurrency, IShippingLine, IApplicationState, getOrderInitialData, addPayment, batchRequest, IBatchableRequest, estimateShippingLines, IAddress, IApiSuccessResponse, getLineItems, getCustomer } from '@boldcommerce/checkout-frontend-library';
+import { OnApproveActions, OnApproveData, OnShippingChangeActions, OnShippingChangeData } from '@paypal/paypal-js';
+import { displayError, enableDisableSection, orderProcessing } from 'src/actions';
+import { getPaypalNameSpace, setPaypalNameSpace, paypalOnClick } from 'src/paypal';
+import { IBraintreeClient, IBraintreePaypalCheckoutInstance } from 'src/types';
+import { getTotals, loadJS } from 'src/utils';
+import { showPaymentMethodTypes } from 'src/variables';
+import { braintreeOnLoadClient } from './braintreeOnLoadClient';
+import { getBraintreeJsUrls } from './getBraintreeJsUrls';
+import { getBraintreeClient } from './manageBraintreeState';
+
+// TODO move this interface to frontend library
+export interface IExpressPayBraintreePaypal {
+    is_paylater_enabled: boolean;
+    is_test: 0 | 1;
+    properties: {
+        public_id: string;
+        commit: boolean;
+    };
+    tokenization_key: string;
+    type: 'braintree-paypal';
+    validation_schema: unknown[];
+}
+
+interface IEstimateShippingLinesResponse {
+    // Yes this is the correct key
+    'shipping options': IShippingLine[];
+    application_state: IApplicationState;
+}
+
+const handleCreateOrder = (paypalCheckout: IBraintreePaypalCheckoutInstance) => async (): Promise<string> => {
+    const totals = getTotals();
+
+    return paypalCheckout.createPayment({
+        currency: getCurrency().iso_code,
+        flow: 'checkout',
+        amount: totals.totalAmountDue / 100,
```

Comment:
Should we have a currency helper here? Is this going to cause off by one cent issues?

Reply:
Fixed